docs/UsingCppInterOp.rst

```diff
@@ -67,33 +67,204 @@ that it was designed for.
 
 Further Enhancing the Dynamic/Automatic bindings in CPPYY
 =========================================================
+
+CppInterOp ties into the following major enhancements in cppyy:
+
+-  **Reducing dependencies** of cppyy to minimize code bloat and make it
+   faster.
+
+-  A **cppyy-Style Library** (i.e., CppInterOp) that enables interoperability
+   with C++ code, bringing the speed and efficiency of C++ to simpler, more
+   interactive languages like Python.
+
+**Reducing Dependencies**
+
+Recent work done on cppyy has been focused on removing unnecessary
+dependencies on domain-specific infrastructure (e.g., the ROOT
+framework). The idea was to convert the cppyy-backend to use Cling
+directly (instead of ROOT meta), and then use it in cppyy.
+
+Only a small set of APIs are needed to connect to the interpreter, since
+other APIs are already available in the standard compiler. This is what
+led to the creation of CppInterOp (a library of helper functions), that
+helped extract out things that were unnecessary for cppyy, etc.
+
+The API surface is now incomparably smaller and simpler than what it
+used to be.
+
+**cppyy-Style Library**
+
 The main use case for CppInterOp is the CPPYY service. CPPYY is an
 automatic run-time bindings generator for Python and C++, and supports a wide
 range of C++ features (e.g., template instantiation). It operates on demand and
 generates only what is necessary. It requires a compiler (Cling or Clang-REPL).
 that can be available during programme runtime.
 
+**LLVM Integration**
+
 Once CppInterOp is integrated with LLVM's Clang-REPL component (that can then
 be used as a runtime compiler for CPPYY), it will further enhance CPPYY's
 performance in the following ways:
 
 
-**Simpler codebase:** The removal of string parsing logic will lead to a simpler
-code base.
+- *Simpler codebase:* The removal of string parsing logic will lead to a
+  simpler code base.
+
+- *LLVM Integration:* The CppInterOp interfaces will be part of the LLVM
```

Comment:
seems circular in logic

Reply:
rephrased:
- *Built into the LLVM toolchain:* The CppInterOp interfaces will be part of
  the LLVM toolchain (as part of Clang-REPL).